contracts/modules/assessment/Assessment.sol

```diff
@@ -2,515 +2,365 @@
+    groupIds = _groupsForAssessor[assessorMemberId].values();
   }
 
-  /// @dev Returns the number of assessments.
-  function getAssessmentsCount() external override view returns (uint) {
-    return assessments.length;
-  }
+  function getGroupsData(uint[] calldata groupIds) override external view returns (AssessmentGroupView[] memory groups) {
+    uint length = groupIds.length;
+    groups = new AssessmentGroupView[](length);
+
+    for (uint i = 0; i < length; i++) {
+      uint groupId = groupIds[i];
+      groups[i] = AssessmentGroupView({
+        id: groupId,
+        ipfsMetadata: _groupsMetadata[groupId],
+        assessors: _groups[groupId].values()
+      });
+    }
 
-  /// @dev Returns only the poll from the assessment struct to make only one SLOAD. Is used by
-  /// other contracts.
-  ///
-  /// @param assessmentId  The index of the assessment
-  function getPoll(uint assessmentId) external override view returns (Poll memory) {
-    return assessments[assessmentId].poll;
+    return groups;
   }
 
-  /// Returns all pending rewards, the withdrawable amount and the index until which they can be
-  /// withdrawn.
-  ///
-  /// @param staker  The address of the staker
-  function getRewards(address staker) external override view returns (
-    uint totalPendingAmountInNXM,
-    uint withdrawableAmountInNXM,
-    uint withdrawableUntilIndex
-  ) {
-
-    uint104 rewardsWithdrawableFromIndex = stakeOf[staker].rewardsWithdrawableFromIndex;
-    Vote memory vote;
-    Assessment memory assessment;
-    uint voteCount = votesOf[staker].length;
-    bool hasReachedUnwithdrawableReward = false;
-
-    for (uint i = rewardsWithdrawableFromIndex; i < voteCount; i++) {
-      vote = votesOf[staker][i];
-      assessment = assessments[vote.assessmentId];
-
-      // If hasReachedUnwithdrawableReward is true, skip and continue calculating the pending total
-      // rewards.
-      if (
-        !hasReachedUnwithdrawableReward &&
-        uint(assessment.poll.end) + PAYOUT_COOLDOWN >= block.timestamp
-      ) {
-        hasReachedUnwithdrawableReward = true;
-        // Store the index of the vote until which rewards can be withdrawn.
-        withdrawableUntilIndex = i;
-        // Then, also store the pending total value that can be withdrawn until this index.
-        withdrawableAmountInNXM = totalPendingAmountInNXM;
-      }
-
-      totalPendingAmountInNXM += uint(assessment.totalRewardInNXM) * uint(vote.stakedAmount) /
-        (uint(assessment.poll.accepted) + uint(assessment.poll.denied));
+  function setAssessmentDataForProductTypes(
+    uint[] calldata productTypeIds,
+    uint cooldownPeriod,
+    uint groupId
+  ) override external onlyContracts(C_GOVERNOR) {
+    uint length = productTypeIds.length;
+    for (uint i = 0; i < length; i++) {
+      _assessmentData[productTypeIds[i]] = AssessmentData({
+        assessingGroupId: groupId.toUint16(),
```

Comment:
1. the group id is not validated.
2. what happens if we change the group while there are assessments that are in progress?

Reply:
on (2), I guess I've just figured it out - you copy the id when starting an assessment so there should be no issue.